editor/components/rich-text/format-toolbar/index.js

```diff
@@ -93,11 +93,11 @@ class FormatToolbar extends Component {
 		}
 	}
 
-	componentWillReceiveProps( nextProps ) {
-		if ( this.props.selectedNodeId !== nextProps.selectedNodeId ) {
+	componentDidUpdate( prevProps ) {
```

Comment:
In this case we are just setting the state so getDerivedStateFromProps sounds like a good git. The downside is that we will need to keep a reference to selectedNodeId in state but I think it is worth it to avoid double render calls.

Reply:
According to the React rfcs mirroring the values in state sounds like the way to go: https://github.com/reactjs/rfcs/blob/master/text/0006-static-lifecycle-methods.md#state-derived-from-propsstate

Reply:
But we can't  because we use the previous props and since `getDerivedStateFromProps` is static we can't access those.

Reply:
We can save the previous selectedNodeId in the state. And then in getDerivedStateFromProps we compare the new prop with the previous (saved on the state), the same they suggest in https://github.com/reactjs/rfcs/blob/master/text/0006-static-lifecycle-methods.md#state-derived-from-propsstate.